src/pages/SignUpPage/SignUpPage.js

```diff
@@ -0,0 +1,131 @@
+import React from "react";
+import { TextField, Button, FormControlLabel, Checkbox } from "@material-ui/core";
+import {Alert, AlertTitle} from '@material-ui/lab/';
+import {Favorite, FavoriteBorder} from '@material-ui/icons';
+import "./signUp.styles.css";
+import Input from 'react-phone-number-input';
+import 'react-phone-number-input/style.css'
+import { useState } from "react";
+
+
+const SignUpPage = () => {
+   const [phone, setPhone] = useState('')
+   const [name, setName] = useState('');
+   const [email, setEmail] = useState('');
+   const [password, setPassword] = useState('');
+   const data = {
+       "name": name,
+       "email": email,
+       "password": password,
+       "phone": phone,
+   };
+   const [checked, setChecked] = useState(false)
+   const [alert, setAlert] = useState(false);
+   const [isSubmitted, setIsSubmitted] = useState(false);
+   const [otp, setOtp] = useState('');
+   const [success, setSuccess] = useState(false)
+
+   const handleName = (e) => {
+       setName(e.target.value)
+   }
+
+   const handleEmail = (e) => {
+       setEmail(e.target.value)
+   }
+
+   const handlePassword = (e) => {
+       setPassword(e.target.value)
+   }
+
+   const handleCheck = (e) => {
+       if (e.target.checked) {
+           setChecked(true)
+       }
+   }
+
+   const handleSubmit = (e) => {
+       e.preventDefault();
+       if(!name || !email || !email.includes("@") || !password || !phone || phone.length !== 12 || checked === false) {
+         setAlert(true)
+       } else {
+        setAlert(false);
+        setIsSubmitted(true)
+       }
+   }
+
+   const handleOtp = (e) => {
+       setOtp(e.target.value)
+   }
+
+   const handleSubmitOTP = (e) => {
+       e.preventDefault();
+       if (otp) {
+        setAlert(false);
+        setSuccess(true)}
+       else {
+           setAlert(true);
+           setSuccess(false);
+       }
+   }
+
+
+   console.log("user data", data)
+    return (
+        <>
+       
+        <Alert severity="error" style={{display: alert ? "block" : "none"}}>
+           <AlertTitle>Error</AlertTitle> {isSubmitted ? "This field is required." : "Please fill down the valid content in all below fields and check our Temrs & Conditions."}
+            — <strong>Check again!</strong>
+        </Alert>
+
+        <Alert severity="success" style={{display: success ? "block": "none"}}>
+        <AlertTitle>Success</AlertTitle>
+        Your account has been created successfully — <strong><a href="/login">You may log in right here.</a></strong>
+      </Alert>
+
+    { isSubmitted ? (
+         <form  className="formSignUp" noValidate autoComplete="off" onSubmit={handleSubmitOTP}>
```

Comment:
just a quick code thing, i think it'd be easier to follow if you split this up into smaller components. 

so code for this whole component could be something like:

```
return (
  // ... alert stuff which probably needs to be edited a bit 

  {{ isSubmitted ? <OTPForm onSubmit={handleSubmitOTP} onChange={handleOTP}  : <SignupForm onSubmit={handleSubmit}/> }}
)
```

Then in separate files, you define each of these components. That helps keep the logic separate. One thing that gets messy though is that we're passing handlers, so ... redux to the rescue, as you wouldn't even have to pass the props (redux store/container/actions would take care of it).